swift/ql/src/queries/Security/CWE-825/UnsafePointerEscapesClosure.ql

```diff
@@ -0,0 +1,94 @@
+/**
+ * @name Unsafe pointer escapes closure
+ * @description Certain functions pass a low-level pointer to a closure. If this pointer outlives the closure, unpredictable results may occur.
+ * @kind path-problem
+ * @id swift/unsafe-pointer-escapes-closure
+ * @tags security
+ *       external/cwe/cwe-825
+ */
+
+import swift
+import codeql.swift.dataflow.DataFlow
+import Flow::PathGraph
+
+module Conf implements DataFlow::StateConfigSig {
+  class FlowState = Callable;
+
+  additional predicate isUnsafePointerCall(CallExpr call) {
+    call.getStaticTarget()
+    .hasName([
+        "withUnsafeBytes(_:)", "withCString(_:)", "withUnsafeMutableBytes(_:)",
+        "withContiguousMutableStorageIfAvailable(_:)", "withContiguousStorageIfAvailable(_:)",
+        "withUTF8(_:)", "withUnsafeBufferPointer(_:)", "withUnsafeBufferPointer(_:)"
```

Comment:
Good list.  You've duplicated `withUnsafeBufferPointer`.
```suggestion
        "withUTF8(_:)", "withUnsafeBufferPointer(_:)", "withUnsafeMutableBufferPointer(_:)"
```

I think we could add [`withMemoryRebound(to:capacity:_:)`](https://developer.apple.com/documentation/swift/unsaferawpointer/withmemoryrebound(to:capacity:_:)) as well (the 2nd argument takes a similar type of closure).